fe/src/components/RatingsHistory/RatingsHistoryChart.re

```diff
@@ -4,13 +4,25 @@ open EloTypes;
 
 let component = statelessComponent("RatingsHistoryChart");
 
+let sumRatingWithDiff = (rating: ratingHistory): int => {
+  rating.userRating + rating.ratingDiff;
+};
+
+let calculateActualRating = (ratingsHistory: list(ratingHistory)): int => {
+  ratingsHistory |> List.hd |> sumRatingWithDiff;
+};
+
 let make = (~ratingsHistory, _children) => {
   ...component,
   render: _self => {
+    let actualRating = ratingsHistory |> calculateActualRating;
+    let actualRatingObj = [{"name": "actual", "rating": actualRating}];
+
     let data =
       ratingsHistory
-      |> List.rev
       |> List.map(item => {"name": item.date, "rating": item.userRating})
+      |> List.append(actualRatingObj)
```

Comment:
@ktrzos 
Generalnie z tego co czytałem to append - czyli dodawanie na końcu jest dosyć kosztowne, że te listy są zaprojektowane żeby łatwo się dodawało na początku. Więc może lepiej najpierw zrobić List.rev a potem dodać obiekt na początku za pomocą trzech kropek

Reply:
Zmieniłem, i przedstawiam w kodzie dwie propozycje. **Pierwsza** z nich jest dokonana przy użyciu całkowitego pipeline i `.rev_append()`, która według dokumentacji dokonuje operacji trybem "tail-recursive", czyli niby bardziej optymalnym. **Drugi** przypadek zrobiłem nieco inaczej, i bez użycia operatora `...` (bo mi go nawet usuwało). Nie wiem czy jest dobrze - nie mam innych pomysłów na to.